// @neynar/nodejs-sdk/build/neynar-api/v2/openapi-farcaster/apis/fname-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster API V2
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FnameApi = exports.FnameApiFactory = exports.FnameApiFp = exports.FnameApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * FnameApi - axios parameter creator
 * @export
 */
const FnameApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Check if a given fname is available
         * @summary Check if a given fname is available
         * @param {string} apiKey API key required for authentication.
         * @param {string} fname
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        fnameAvailability: async (apiKey, fname, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('fnameAvailability', 'apiKey', apiKey);
            // verify required parameter 'fname' is not null or undefined
            (0, common_1.assertParamExists)('fnameAvailability', 'fname', fname);
            const localVarPath = `/farcaster/fname/availability`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (fname !== undefined) {
                localVarQueryParameter['fname'] = fname;
            }
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.FnameApiAxiosParamCreator = FnameApiAxiosParamCreator;
/**
 * FnameApi - functional programming interface
 * @export
 */
const FnameApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.FnameApiAxiosParamCreator)(configuration);
    return {
        /**
         * Check if a given fname is available
         * @summary Check if a given fname is available
         * @param {string} apiKey API key required for authentication.
         * @param {string} fname
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async fnameAvailability(apiKey, fname, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.fnameAvailability(apiKey, fname, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
exports.FnameApiFp = FnameApiFp;
/**
 * FnameApi - factory interface
 * @export
 */
const FnameApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.FnameApiFp)(configuration);
    return {
        /**
         * Check if a given fname is available
         * @summary Check if a given fname is available
         * @param {string} apiKey API key required for authentication.
         * @param {string} fname
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        fnameAvailability(apiKey, fname, options) {
            return localVarFp.fnameAvailability(apiKey, fname, options).then((request) => request(axios, basePath));
        },
    };
};
exports.FnameApiFactory = FnameApiFactory;
/**
 * FnameApi - object-oriented interface
 * @export
 * @class FnameApi
 * @extends {BaseAPI}
 */
class FnameApi extends base_1.BaseAPI {
    /**
     * Check if a given fname is available
     * @summary Check if a given fname is available
     * @param {string} apiKey API key required for authentication.
     * @param {string} fname
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FnameApi
     */
    fnameAvailability(apiKey, fname, options) {
        return (0, exports.FnameApiFp)(this.configuration).fnameAvailability(apiKey, fname, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.FnameApi = FnameApi;
