// @neynar/nodejs-sdk/build/neynar-api/v2/openapi-farcaster/apis/reaction-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster API V2
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReactionApi = exports.ReactionApiFactory = exports.ReactionApiFp = exports.ReactionApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * ReactionApi - axios parameter creator
 * @export
 */
const ReactionApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Delete a reaction (like or recast) to a given cast \\ (In order to delete a reaction `signer_uuid` must be approved)
         * @summary Delete a reaction
         * @param {string} apiKey API key required for authentication.
         * @param {ReactionReqBody} reactionReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteReaction: async (apiKey, reactionReqBody, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('deleteReaction', 'apiKey', apiKey);
            // verify required parameter 'reactionReqBody' is not null or undefined
            (0, common_1.assertParamExists)('deleteReaction', 'reactionReqBody', reactionReqBody);
            const localVarPath = `/farcaster/reaction`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'DELETE' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(reactionReqBody, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Post a reaction (like or recast) to a given cast \\ (In order to post a reaction `signer_uuid` must be approved)
         * @summary Posts a reaction
         * @param {string} apiKey API key required for authentication.
         * @param {ReactionReqBody} reactionReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        postReaction: async (apiKey, reactionReqBody, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('postReaction', 'apiKey', apiKey);
            // verify required parameter 'reactionReqBody' is not null or undefined
            (0, common_1.assertParamExists)('postReaction', 'reactionReqBody', reactionReqBody);
            const localVarPath = `/farcaster/reaction`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(reactionReqBody, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Fetches reactions for a given cast
         * @summary Fetches reactions for a given cast
         * @param {string} apiKey API key required for authentication.
         * @param {string} hash
         * @param {string} types Customize which reaction types the request should search for. This is a comma-separated string that can include the following values: \&#39;likes\&#39; and \&#39;recasts\&#39;. By default api returns both. To select multiple types, use a comma-separated list of these values.
         * @param {number} [limit] Number of results to retrieve (default 25, max 100)
         * @param {string} [cursor] Pagination cursor.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        reactionsCast: async (apiKey, hash, types, limit, cursor, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('reactionsCast', 'apiKey', apiKey);
            // verify required parameter 'hash' is not null or undefined
            (0, common_1.assertParamExists)('reactionsCast', 'hash', hash);
            // verify required parameter 'types' is not null or undefined
            (0, common_1.assertParamExists)('reactionsCast', 'types', types);
            const localVarPath = `/farcaster/reactions/cast`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (hash !== undefined) {
                localVarQueryParameter['hash'] = hash;
            }
            if (types !== undefined) {
                localVarQueryParameter['types'] = types;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (cursor !== undefined) {
                localVarQueryParameter['cursor'] = cursor;
            }
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Fetches reactions for a given user
         * @summary Fetches reactions for a given user
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {ReactionsType} type Type of reaction to fetch (likes or recasts or all)
         * @param {number} [limit] Number of results to retrieve (default 25, max 100)
         * @param {string} [cursor] Pagination cursor.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        reactionsUser: async (apiKey, fid, type, limit, cursor, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('reactionsUser', 'apiKey', apiKey);
            // verify required parameter 'fid' is not null or undefined
            (0, common_1.assertParamExists)('reactionsUser', 'fid', fid);
            // verify required parameter 'type' is not null or undefined
            (0, common_1.assertParamExists)('reactionsUser', 'type', type);
            const localVarPath = `/farcaster/reactions/user`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (type !== undefined) {
                localVarQueryParameter['type'] = type;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (cursor !== undefined) {
                localVarQueryParameter['cursor'] = cursor;
            }
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.ReactionApiAxiosParamCreator = ReactionApiAxiosParamCreator;
/**
 * ReactionApi - functional programming interface
 * @export
 */
const ReactionApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.ReactionApiAxiosParamCreator)(configuration);
    return {
        /**
         * Delete a reaction (like or recast) to a given cast \\ (In order to delete a reaction `signer_uuid` must be approved)
         * @summary Delete a reaction
         * @param {string} apiKey API key required for authentication.
         * @param {ReactionReqBody} reactionReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteReaction(apiKey, reactionReqBody, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteReaction(apiKey, reactionReqBody, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Post a reaction (like or recast) to a given cast \\ (In order to post a reaction `signer_uuid` must be approved)
         * @summary Posts a reaction
         * @param {string} apiKey API key required for authentication.
         * @param {ReactionReqBody} reactionReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async postReaction(apiKey, reactionReqBody, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.postReaction(apiKey, reactionReqBody, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Fetches reactions for a given cast
         * @summary Fetches reactions for a given cast
         * @param {string} apiKey API key required for authentication.
         * @param {string} hash
         * @param {string} types Customize which reaction types the request should search for. This is a comma-separated string that can include the following values: \&#39;likes\&#39; and \&#39;recasts\&#39;. By default api returns both. To select multiple types, use a comma-separated list of these values.
         * @param {number} [limit] Number of results to retrieve (default 25, max 100)
         * @param {string} [cursor] Pagination cursor.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async reactionsCast(apiKey, hash, types, limit, cursor, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.reactionsCast(apiKey, hash, types, limit, cursor, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Fetches reactions for a given user
         * @summary Fetches reactions for a given user
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {ReactionsType} type Type of reaction to fetch (likes or recasts or all)
         * @param {number} [limit] Number of results to retrieve (default 25, max 100)
         * @param {string} [cursor] Pagination cursor.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async reactionsUser(apiKey, fid, type, limit, cursor, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.reactionsUser(apiKey, fid, type, limit, cursor, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
exports.ReactionApiFp = ReactionApiFp;
/**
 * ReactionApi - factory interface
 * @export
 */
const ReactionApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.ReactionApiFp)(configuration);
    return {
        /**
         * Delete a reaction (like or recast) to a given cast \\ (In order to delete a reaction `signer_uuid` must be approved)
         * @summary Delete a reaction
         * @param {string} apiKey API key required for authentication.
         * @param {ReactionReqBody} reactionReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteReaction(apiKey, reactionReqBody, options) {
            return localVarFp.deleteReaction(apiKey, reactionReqBody, options).then((request) => request(axios, basePath));
        },
        /**
         * Post a reaction (like or recast) to a given cast \\ (In order to post a reaction `signer_uuid` must be approved)
         * @summary Posts a reaction
         * @param {string} apiKey API key required for authentication.
         * @param {ReactionReqBody} reactionReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        postReaction(apiKey, reactionReqBody, options) {
            return localVarFp.postReaction(apiKey, reactionReqBody, options).then((request) => request(axios, basePath));
        },
        /**
         * Fetches reactions for a given cast
         * @summary Fetches reactions for a given cast
         * @param {string} apiKey API key required for authentication.
         * @param {string} hash
         * @param {string} types Customize which reaction types the request should search for. This is a comma-separated string that can include the following values: \&#39;likes\&#39; and \&#39;recasts\&#39;. By default api returns both. To select multiple types, use a comma-separated list of these values.
         * @param {number} [limit] Number of results to retrieve (default 25, max 100)
         * @param {string} [cursor] Pagination cursor.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        reactionsCast(apiKey, hash, types, limit, cursor, options) {
            return localVarFp.reactionsCast(apiKey, hash, types, limit, cursor, options).then((request) => request(axios, basePath));
        },
        /**
         * Fetches reactions for a given user
         * @summary Fetches reactions for a given user
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {ReactionsType} type Type of reaction to fetch (likes or recasts or all)
         * @param {number} [limit] Number of results to retrieve (default 25, max 100)
         * @param {string} [cursor] Pagination cursor.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        reactionsUser(apiKey, fid, type, limit, cursor, options) {
            return localVarFp.reactionsUser(apiKey, fid, type, limit, cursor, options).then((request) => request(axios, basePath));
        },
    };
};
exports.ReactionApiFactory = ReactionApiFactory;
/**
 * ReactionApi - object-oriented interface
 * @export
 * @class ReactionApi
 * @extends {BaseAPI}
 */
class ReactionApi extends base_1.BaseAPI {
    /**
     * Delete a reaction (like or recast) to a given cast \\ (In order to delete a reaction `signer_uuid` must be approved)
     * @summary Delete a reaction
     * @param {string} apiKey API key required for authentication.
     * @param {ReactionReqBody} reactionReqBody
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ReactionApi
     */
    deleteReaction(apiKey, reactionReqBody, options) {
        return (0, exports.ReactionApiFp)(this.configuration).deleteReaction(apiKey, reactionReqBody, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Post a reaction (like or recast) to a given cast \\ (In order to post a reaction `signer_uuid` must be approved)
     * @summary Posts a reaction
     * @param {string} apiKey API key required for authentication.
     * @param {ReactionReqBody} reactionReqBody
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ReactionApi
     */
    postReaction(apiKey, reactionReqBody, options) {
        return (0, exports.ReactionApiFp)(this.configuration).postReaction(apiKey, reactionReqBody, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Fetches reactions for a given cast
     * @summary Fetches reactions for a given cast
     * @param {string} apiKey API key required for authentication.
     * @param {string} hash
     * @param {string} types Customize which reaction types the request should search for. This is a comma-separated string that can include the following values: \&#39;likes\&#39; and \&#39;recasts\&#39;. By default api returns both. To select multiple types, use a comma-separated list of these values.
     * @param {number} [limit] Number of results to retrieve (default 25, max 100)
     * @param {string} [cursor] Pagination cursor.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ReactionApi
     */
    reactionsCast(apiKey, hash, types, limit, cursor, options) {
        return (0, exports.ReactionApiFp)(this.configuration).reactionsCast(apiKey, hash, types, limit, cursor, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Fetches reactions for a given user
     * @summary Fetches reactions for a given user
     * @param {string} apiKey API key required for authentication.
     * @param {number} fid
     * @param {ReactionsType} type Type of reaction to fetch (likes or recasts or all)
     * @param {number} [limit] Number of results to retrieve (default 25, max 100)
     * @param {string} [cursor] Pagination cursor.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ReactionApi
     */
    reactionsUser(apiKey, fid, type, limit, cursor, options) {
        return (0, exports.ReactionApiFp)(this.configuration).reactionsUser(apiKey, fid, type, limit, cursor, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.ReactionApi = ReactionApi;
