// @neynar/nodejs-sdk/build/neynar-api/v2/openapi-farcaster/apis/storage-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster API V2
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageApi = exports.StorageApiFactory = exports.StorageApiFp = exports.StorageApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * StorageApi - axios parameter creator
 * @export
 */
const StorageApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Fetches storage allocations for a given user
         * @summary Fetches storage allocations for a given user
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        storageAllocations: async (apiKey, fid, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('storageAllocations', 'apiKey', apiKey);
            // verify required parameter 'fid' is not null or undefined
            (0, common_1.assertParamExists)('storageAllocations', 'fid', fid);
            const localVarPath = `/farcaster/storage/allocations`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Fetches storage usage for a given user
         * @summary Fetches storage usage for a given user
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        storageUsage: async (apiKey, fid, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('storageUsage', 'apiKey', apiKey);
            // verify required parameter 'fid' is not null or undefined
            (0, common_1.assertParamExists)('storageUsage', 'fid', fid);
            const localVarPath = `/farcaster/storage/usage`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.StorageApiAxiosParamCreator = StorageApiAxiosParamCreator;
/**
 * StorageApi - functional programming interface
 * @export
 */
const StorageApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.StorageApiAxiosParamCreator)(configuration);
    return {
        /**
         * Fetches storage allocations for a given user
         * @summary Fetches storage allocations for a given user
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async storageAllocations(apiKey, fid, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.storageAllocations(apiKey, fid, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Fetches storage usage for a given user
         * @summary Fetches storage usage for a given user
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async storageUsage(apiKey, fid, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.storageUsage(apiKey, fid, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
exports.StorageApiFp = StorageApiFp;
/**
 * StorageApi - factory interface
 * @export
 */
const StorageApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.StorageApiFp)(configuration);
    return {
        /**
         * Fetches storage allocations for a given user
         * @summary Fetches storage allocations for a given user
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        storageAllocations(apiKey, fid, options) {
            return localVarFp.storageAllocations(apiKey, fid, options).then((request) => request(axios, basePath));
        },
        /**
         * Fetches storage usage for a given user
         * @summary Fetches storage usage for a given user
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        storageUsage(apiKey, fid, options) {
            return localVarFp.storageUsage(apiKey, fid, options).then((request) => request(axios, basePath));
        },
    };
};
exports.StorageApiFactory = StorageApiFactory;
/**
 * StorageApi - object-oriented interface
 * @export
 * @class StorageApi
 * @extends {BaseAPI}
 */
class StorageApi extends base_1.BaseAPI {
    /**
     * Fetches storage allocations for a given user
     * @summary Fetches storage allocations for a given user
     * @param {string} apiKey API key required for authentication.
     * @param {number} fid
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StorageApi
     */
    storageAllocations(apiKey, fid, options) {
        return (0, exports.StorageApiFp)(this.configuration).storageAllocations(apiKey, fid, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Fetches storage usage for a given user
     * @summary Fetches storage usage for a given user
     * @param {string} apiKey API key required for authentication.
     * @param {number} fid
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StorageApi
     */
    storageUsage(apiKey, fid, options) {
        return (0, exports.StorageApiFp)(this.configuration).storageUsage(apiKey, fid, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.StorageApi = StorageApi;
