// @neynar/nodejs-sdk/build/neynar-api/v2/openapi-farcaster/apis/webhook-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster API V2
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookApi = exports.WebhookApiFactory = exports.WebhookApiFp = exports.WebhookApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * WebhookApi - axios parameter creator
 * @export
 */
const WebhookApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Delete a webhook
         * @summary Delete a webhook
         * @param {string} apiKey API key required for authentication.
         * @param {WebhookDeleteReqBody} webhookDeleteReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteWebhook: async (apiKey, webhookDeleteReqBody, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('deleteWebhook', 'apiKey', apiKey);
            // verify required parameter 'webhookDeleteReqBody' is not null or undefined
            (0, common_1.assertParamExists)('deleteWebhook', 'webhookDeleteReqBody', webhookDeleteReqBody);
            const localVarPath = `/farcaster/webhook`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'DELETE' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(webhookDeleteReqBody, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Fetch a list of webhooks associated to a user
         * @summary Fetch a list of webhooks associated to a user
         * @param {string} apiKey API key required for authentication.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        fetchWebhooks: async (apiKey, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('fetchWebhooks', 'apiKey', apiKey);
            const localVarPath = `/farcaster/webhook/list`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Fetch a webhook
         * @summary Fetch a webhook
         * @param {string} apiKey API key required for authentication.
         * @param {string} webhookId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        lookupWebhook: async (apiKey, webhookId, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('lookupWebhook', 'apiKey', apiKey);
            // verify required parameter 'webhookId' is not null or undefined
            (0, common_1.assertParamExists)('lookupWebhook', 'webhookId', webhookId);
            const localVarPath = `/farcaster/webhook`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (webhookId !== undefined) {
                localVarQueryParameter['webhook_id'] = webhookId;
            }
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Create a webhook
         * @summary Create a webhook
         * @param {string} apiKey API key required for authentication.
         * @param {WebhookPostReqBody} webhookPostReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        publishWebhook: async (apiKey, webhookPostReqBody, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('publishWebhook', 'apiKey', apiKey);
            // verify required parameter 'webhookPostReqBody' is not null or undefined
            (0, common_1.assertParamExists)('publishWebhook', 'webhookPostReqBody', webhookPostReqBody);
            const localVarPath = `/farcaster/webhook`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(webhookPostReqBody, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Update a webhook
         * @summary Update a webhook
         * @param {string} apiKey API key required for authentication.
         * @param {WebhookPutReqBody} webhookPutReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateWebhook: async (apiKey, webhookPutReqBody, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('updateWebhook', 'apiKey', apiKey);
            // verify required parameter 'webhookPutReqBody' is not null or undefined
            (0, common_1.assertParamExists)('updateWebhook', 'webhookPutReqBody', webhookPutReqBody);
            const localVarPath = `/farcaster/webhook`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'PUT' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(webhookPutReqBody, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Update webhook active status
         * @summary Update webhook active status
         * @param {string} apiKey API key required for authentication.
         * @param {WebhookPatchReqBody} webhookPatchReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateWebhookActiveStatus: async (apiKey, webhookPatchReqBody, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('updateWebhookActiveStatus', 'apiKey', apiKey);
            // verify required parameter 'webhookPatchReqBody' is not null or undefined
            (0, common_1.assertParamExists)('updateWebhookActiveStatus', 'webhookPatchReqBody', webhookPatchReqBody);
            const localVarPath = `/farcaster/webhook`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'PATCH' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(webhookPatchReqBody, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.WebhookApiAxiosParamCreator = WebhookApiAxiosParamCreator;
/**
 * WebhookApi - functional programming interface
 * @export
 */
const WebhookApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.WebhookApiAxiosParamCreator)(configuration);
    return {
        /**
         * Delete a webhook
         * @summary Delete a webhook
         * @param {string} apiKey API key required for authentication.
         * @param {WebhookDeleteReqBody} webhookDeleteReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteWebhook(apiKey, webhookDeleteReqBody, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteWebhook(apiKey, webhookDeleteReqBody, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Fetch a list of webhooks associated to a user
         * @summary Fetch a list of webhooks associated to a user
         * @param {string} apiKey API key required for authentication.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async fetchWebhooks(apiKey, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.fetchWebhooks(apiKey, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Fetch a webhook
         * @summary Fetch a webhook
         * @param {string} apiKey API key required for authentication.
         * @param {string} webhookId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async lookupWebhook(apiKey, webhookId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.lookupWebhook(apiKey, webhookId, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Create a webhook
         * @summary Create a webhook
         * @param {string} apiKey API key required for authentication.
         * @param {WebhookPostReqBody} webhookPostReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async publishWebhook(apiKey, webhookPostReqBody, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.publishWebhook(apiKey, webhookPostReqBody, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Update a webhook
         * @summary Update a webhook
         * @param {string} apiKey API key required for authentication.
         * @param {WebhookPutReqBody} webhookPutReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateWebhook(apiKey, webhookPutReqBody, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateWebhook(apiKey, webhookPutReqBody, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Update webhook active status
         * @summary Update webhook active status
         * @param {string} apiKey API key required for authentication.
         * @param {WebhookPatchReqBody} webhookPatchReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateWebhookActiveStatus(apiKey, webhookPatchReqBody, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateWebhookActiveStatus(apiKey, webhookPatchReqBody, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
exports.WebhookApiFp = WebhookApiFp;
/**
 * WebhookApi - factory interface
 * @export
 */
const WebhookApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.WebhookApiFp)(configuration);
    return {
        /**
         * Delete a webhook
         * @summary Delete a webhook
         * @param {string} apiKey API key required for authentication.
         * @param {WebhookDeleteReqBody} webhookDeleteReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteWebhook(apiKey, webhookDeleteReqBody, options) {
            return localVarFp.deleteWebhook(apiKey, webhookDeleteReqBody, options).then((request) => request(axios, basePath));
        },
        /**
         * Fetch a list of webhooks associated to a user
         * @summary Fetch a list of webhooks associated to a user
         * @param {string} apiKey API key required for authentication.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        fetchWebhooks(apiKey, options) {
            return localVarFp.fetchWebhooks(apiKey, options).then((request) => request(axios, basePath));
        },
        /**
         * Fetch a webhook
         * @summary Fetch a webhook
         * @param {string} apiKey API key required for authentication.
         * @param {string} webhookId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        lookupWebhook(apiKey, webhookId, options) {
            return localVarFp.lookupWebhook(apiKey, webhookId, options).then((request) => request(axios, basePath));
        },
        /**
         * Create a webhook
         * @summary Create a webhook
         * @param {string} apiKey API key required for authentication.
         * @param {WebhookPostReqBody} webhookPostReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        publishWebhook(apiKey, webhookPostReqBody, options) {
            return localVarFp.publishWebhook(apiKey, webhookPostReqBody, options).then((request) => request(axios, basePath));
        },
        /**
         * Update a webhook
         * @summary Update a webhook
         * @param {string} apiKey API key required for authentication.
         * @param {WebhookPutReqBody} webhookPutReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateWebhook(apiKey, webhookPutReqBody, options) {
            return localVarFp.updateWebhook(apiKey, webhookPutReqBody, options).then((request) => request(axios, basePath));
        },
        /**
         * Update webhook active status
         * @summary Update webhook active status
         * @param {string} apiKey API key required for authentication.
         * @param {WebhookPatchReqBody} webhookPatchReqBody
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateWebhookActiveStatus(apiKey, webhookPatchReqBody, options) {
            return localVarFp.updateWebhookActiveStatus(apiKey, webhookPatchReqBody, options).then((request) => request(axios, basePath));
        },
    };
};
exports.WebhookApiFactory = WebhookApiFactory;
/**
 * WebhookApi - object-oriented interface
 * @export
 * @class WebhookApi
 * @extends {BaseAPI}
 */
class WebhookApi extends base_1.BaseAPI {
    /**
     * Delete a webhook
     * @summary Delete a webhook
     * @param {string} apiKey API key required for authentication.
     * @param {WebhookDeleteReqBody} webhookDeleteReqBody
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof WebhookApi
     */
    deleteWebhook(apiKey, webhookDeleteReqBody, options) {
        return (0, exports.WebhookApiFp)(this.configuration).deleteWebhook(apiKey, webhookDeleteReqBody, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Fetch a list of webhooks associated to a user
     * @summary Fetch a list of webhooks associated to a user
     * @param {string} apiKey API key required for authentication.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof WebhookApi
     */
    fetchWebhooks(apiKey, options) {
        return (0, exports.WebhookApiFp)(this.configuration).fetchWebhooks(apiKey, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Fetch a webhook
     * @summary Fetch a webhook
     * @param {string} apiKey API key required for authentication.
     * @param {string} webhookId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof WebhookApi
     */
    lookupWebhook(apiKey, webhookId, options) {
        return (0, exports.WebhookApiFp)(this.configuration).lookupWebhook(apiKey, webhookId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Create a webhook
     * @summary Create a webhook
     * @param {string} apiKey API key required for authentication.
     * @param {WebhookPostReqBody} webhookPostReqBody
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof WebhookApi
     */
    publishWebhook(apiKey, webhookPostReqBody, options) {
        return (0, exports.WebhookApiFp)(this.configuration).publishWebhook(apiKey, webhookPostReqBody, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Update a webhook
     * @summary Update a webhook
     * @param {string} apiKey API key required for authentication.
     * @param {WebhookPutReqBody} webhookPutReqBody
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof WebhookApi
     */
    updateWebhook(apiKey, webhookPutReqBody, options) {
        return (0, exports.WebhookApiFp)(this.configuration).updateWebhook(apiKey, webhookPutReqBody, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Update webhook active status
     * @summary Update webhook active status
     * @param {string} apiKey API key required for authentication.
     * @param {WebhookPatchReqBody} webhookPatchReqBody
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof WebhookApi
     */
    updateWebhookActiveStatus(apiKey, webhookPatchReqBody, options) {
        return (0, exports.WebhookApiFp)(this.configuration).updateWebhookActiveStatus(apiKey, webhookPatchReqBody, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.WebhookApi = WebhookApi;
